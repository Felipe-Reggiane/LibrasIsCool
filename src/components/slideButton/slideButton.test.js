import React from "react";
import { render, fireEvent } from "@testing-library/react";
import SlideButton from "./slideButton";

import { JSDOM } from "jsdom";

const dom = new JSDOM("<!doctype html><html><body></body></html>");
global.window = dom.window;
global.document = dom.window.document;

jest.mock("react", () => ({
  ...jest.requireActual("react"),
  useContext: jest.fn(),
}));

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useLocation: () => ({
    pathname: "/",
  }),
  useNavigate: () => jest.fn(),
}));

afterEach(() => {
  jest.clearAllMocks();
});
describe("test SlideButton", () => {
  test("SlideButton renders correctly", () => {
    const { asFragment } = render(<SlideButton />);
    expect(asFragment()).toMatchSnapshot();
  });

  test("handle left click", () => {
    const navigate = jest.fn();
    jest
      .spyOn(require("react-router-dom"), "useNavigate")
      .mockReturnValue(navigate);
    jest
      .spyOn(require("react-router-dom"), "useLocation")
      .mockReturnValue({ pathname: "/buttons" });
    const { getByText } = render(<SlideButton />);

    // Simulate a click on the "Digitar texto" button
    fireEvent.click(getByText("Digitar texto"));
    expect(navigate).toHaveBeenCalledWith("/home");
  });

  test("handle right click", () => {
    const navigate = jest.fn();
    jest
      .spyOn(require("react-router-dom"), "useNavigate")
      .mockReturnValue(navigate);
    jest
      .spyOn(require("react-router-dom"), "useLocation")
      .mockReturnValue({ pathname: "/home" });
    const { getByText } = render(<SlideButton />);

    fireEvent.click(getByText("Bot√µes"));
    expect(navigate).toHaveBeenCalledWith("/buttons");
  });
});
