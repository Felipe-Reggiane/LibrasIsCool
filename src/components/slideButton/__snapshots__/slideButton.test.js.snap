// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SlideButton renders correctly 1`] = `
<DocumentFragment>
  <div
    class="slide-button-container"
  >
    <div
      class="slide-button-option "
    >
      Digitar texto
    </div>
    <div
      class="slide-button-option "
    >
      Botões
    </div>
  </div>
</DocumentFragment>
`;

exports[`test SlideButton SlideButton renders correctly 1`] = `
<DocumentFragment>
  <div
    class="slide-button-container"
  >
    <div
      class="slide-button-option "
    >
      Digitar texto
    </div>
    <div
      class="slide-button-option "
    >
      Botões
    </div>
  </div>
</DocumentFragment>
`;

exports[`test left e right buttom SlideButton renders correctly 1`] = `
<DocumentFragment>
  <div
    class="slide-button-container"
  >
    <div
      class="slide-button-option "
    >
      Digitar texto
    </div>
    <div
      class="slide-button-option "
    >
      Botões
    </div>
  </div>
</DocumentFragment>
`;
